app/agents_v2/primary_agent/prompts/response_formatter.py

```
@@ -584,4 +584,91 @@ def _generate_contextual_pro_tips(cls, issue: str, emotion: EmotionalState) -> s
         return '\n'.join(f"- {tip}" for tip in all_tips[:3])  # Return up to 3 tips
 
         # Apply formatting rules
-        return cls.apply_mandatory_formatting(enhanced_response)
\ No newline at end of file
+        return cls.apply_mandatory_formatting(enhanced_response)
```

Ruff (0.12.2)

Undefined name `enhanced_response`

(F821)